lib/string/strchr/strchrcnt.h

```diff
@@ -0,0 +1,35 @@
+// SPDX-FileCopyrightText: 2024, Alejandro Colomar <alx@kernel.org>
+// SPDX-License-Identifier: BSD-3-Clause
+
+
+#ifndef SHADOW_INCLUDE_LIB_STRING_STRCHR_STRCHRCNT_H_
+#define SHADOW_INCLUDE_LIB_STRING_STRCHR_STRCHRCNT_H_
+
+
+#include <config.h>
+
+#include <stddef.h>
+#include <string.h>
+
+#include "attr.h"
+
+
+ATTR_STRING(1)
+inline size_t strchrcnt(const char *s, char c);
+
+
+inline size_t
+strchrcnt(const char *s, char c)
+{
+	size_t  n = 0;
+
+	for (; strcmp(s, "") != 0; s++) {
```

Comment:
Why do you use `strcmp(s, "") != 0` here instead of `*s != '\0'`?

Reply:
Hi @stoeckmann !

My plan is to eventually change that to use streq().  I have a patch set written for that already, but have it queued to after some of the current patches are merged.

```c
/* Return true if s1 and s2 compare equal.  */
inline bool
streq(const char *s1, const char *s2)
{
       return strcmp(s1, s2) == 0;
}
```

Then the plan is that everything that can use streq() will use it.  In this case, it would end up being:

```c
for (; !streq(s, ""); s++) {
```

The rationale being that:

-  It's protected by _FORTIFY_SOURCE
-  It's easier to grep for this pattern than for direct byte comparisons.
-  There are less moving parts, so less chance for typos and errors.

I understand it's a bit unidiomatic, so difficult to get used to it, but I think it will be a good investment in the long term.

It's similar to the change done in 59e5eef38f899cf00925f0117e3173566b14a5fc, but this time for reads instead of writes.

Reply:
Is it better now?